tests/PHPStan/Rules/Comparison/data/bug-7176.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1); // lint >= 8.1
+
+namespace Bug7176;
+
+enum Suit
+{
+	case Hearts;
+	case Diamonds;
+	case Clubs;
+	case Spades;
+}
+
+function test(Suit $x): string {
+	if ($x === Suit::Clubs) {
+		return 'WORKS';
+	}
+	// Suit::Clubs is correctly eliminated from possible values
```

Comment:
Shouldn't you verify these assumptions in a NodeScopeResolverTest?

Reply:
sure, that definitely does not hurt :)